themes/agnoster/agnoster.theme.sh

```diff
@@ -296,6 +297,39 @@ function prompt_virtualenv {
   fi
 }
 
+### pyenv prompt
+function prompt_pyenv {
+  if [[ $PYENV_VIRTUALENV_INIT == 1 ]]; then
+    # Priority is shell > local > global
+    local PYENV_SHELL=$(pyenv shell 2>&1)
+    local PYENV_LOCAL=$(pyenv local 2>&1)
+    local PYENV_GLOBAL=$(pyenv global 2>&1)
+    # pyenv shell is not set
+    if [[ "$PYENV_SHELL" == *"pyenv: no shell-specific version configured"* ]]; then
+      # Check if pyenv local is set
+      if [[ "$PYENV_LOCAL" == *"pyenv: no local version configured"* ]]; then
+        # Neither are set, we're using the global version
+        local PYENV_VERSION=$PYENV_GLOBAL
+      else
+        local PYENV_VERSION=$PYENV_LOCAL
+      fi
+    else
+      local PYENV_VERSION=$PYENV_SHELL
+    fi
```

Comment:
I'm not familiar with `pyenv`, but couldn't these codes be simplified? For example,

```bash
local PYENV_VERSION=$(pyenv shell 2>/dev/null || pyenv local 2>/dev/null || pyenv global 2>/dev/null)
```

Reply:
Ah, let me test that, that might be a lot more efficient.

Reply:
Tried and it doesn't work. It seems there's something going on that makes `pyenv shell` exit 0 when it should exit 1 (and the source code says it does, yet it doesn't), but that gave me a hint of how to check and compact the test.

Reply:
Thanks for trying to make it simpler! I'm satisfied with the updated version.